lib/src/select/Select.tsx

```diff
@@ -292,27 +288,27 @@ const DxcSelect = React.forwardRef<RefType, SelectPropsType>(
       }
     };
 
-    const handleSearchIOnChange = (event) => {
+    const handleSearchIOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
       setSearchValue(event.target.value);
       changeVisualFocusIndex(-1);
       openOptions();
     };
 
-    const handleClearOptionsActionOnClick = (event) => {
+    const handleClearOptionsActionOnClick = (event: React.MouseEvent) => {
```

Comment:
I think we should type the MouseEvent like this: `React.MouseEvent<HTMLButtonElement>`.